packages/create-mc-app/bin/create-mc-app.js

```diff
@@ -1,16 +1,66 @@
 #!/usr/bin/env node
 
-const steps = require('../src/steps');
-const logger = require('../src/logger');
-
-try {
-  const options = steps.parseArguments();
-  steps.downloadTemplate(options);
-  steps.updatePackageInfo(options);
-  steps.installDependencies(options);
-  steps.success(options);
-  process.exit(0);
-} catch (error) {
-  logger.error(error);
-  process.exit(1);
+const mri = require('mri');
+const Listr = require('listr');
+const { shouldUseYarn } = require('../src/utils');
+const tasks = require('../src/tasks');
+const hintOutdatedVersion = require('../src/hint-outdated-version');
+const parseArguments = require('../src/parse-arguments');
+const pkg = require('../package.json');
+
+const currentVersion = pkg.version;
+
+async function execute() {
+  const flags = mri(process.argv.slice(2), { alias: { help: ['h'] } });
+  const commands = flags._;
+
+  if (commands.length === 0 || (flags.help && commands.length === 0)) {
+    console.log(`
+  Usage: create-mc-app [project-directory] [flags]
+
+  Displays help information.
+
+  Options:
+
+    --template <name>                (optional) The name of the template to install [default "starter"]
+                                     Available options: ["starter"]
+    --template-version <version>     (optional) The version of the template to install [default "master"]
+    `);
+    process.exit(0);
+  }
+  console.log(`Version: v${currentVersion}`);
+  hintOutdatedVersion(currentVersion);
+  console.log('');
+  const options = parseArguments(flags);
+
+  const taskList = new Listr([
```

Comment:
Most of the changes are related to "moving things around a bit". The previous steps now are called tasks and return a task-like object for the Listr